### PR TITLE
Adding a unit test to #22418

### PR DESCRIPTION
**Pull request by caesarxuchao**
ref #22418 
ref #19403
@liggitt @lavalamp @smarterclayton @bgrant0607

### CODE REVIEW

File: pkg/registry/pod/etcd/etcd_test.go [hunk 1]
```diff
@@ -130,6 +133,54 @@ func TestDelete(t *testing.T) {
...
+	_, err = registry.Create(testContext, pod)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	// delete object with grace period 0, storage will return NotFound, but the
+	// registry shouldn't get any error since we ignore the NotFound error.
+	zero := int64(0)
+	opt := &api.DeleteOptions{GracePeriodSeconds: &zero}
+	_, err = registry.Delete(testContext, pod.Name, opt)
```
Comment (liggitt) [inline pkg/registry/pod/etcd/etcd_test.go]: Check the return value to make sure it is a pod with a deletion timestamp, etc, set

Comment (liggitt) [inline pkg/registry/pod/etcd/etcd_test.go]: And add a flag in the test storage to make sure our delete override got called and actually returned the error

Comment (caesarxuchao) [inline pkg/registry/pod/etcd/etcd_test.go]: Thanks. I'm ooo. I'll try to adress these later today.

**Comment by caesarxuchao** [comment]
@liggitt I added the logic to check the deleted pod and make sure the overriding Delete method is called. PTAL. Thank you!

File: pkg/registry/pod/etcd/etcd_test.go [hunk 2]
```diff
@@ -130,6 +133,70 @@ func TestDelete(t *testing.T) {
...
+	called := false
+	registry, server := newFailDeleteStorage(t, &called)
+	defer server.Terminate(t)
+
+	// should fail if pod A is not created yet.
+	_, err := registry.Delete(testContext, pod.Name, nil)
+	if !errors.IsNotFound(err) {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
```
Comment (liggitt) [inline pkg/registry/pod/etcd/etcd_test.go]: won't this also set `called` to true here? check and reset it before making the graceful=0 delete call

Comment (caesarxuchao) [inline pkg/registry/pod/etcd/etcd_test.go]: Ping @liggitt :)

Comment (smarterclayton) [inline pkg/registry/pod/etcd/etcd_test.go]: LGTM, although this should probably eventually move to rest/test.  Doesn't have to happen here.

**Pull request merged**